lib/pdf_fill/extras_generator.rb

```diff
@@ -24,15 +24,32 @@ def text?
       @generate_blocks.size.positive?
     end
 
+    def generate_header(pdf)
+      pdf.repeat(:all) do
+        pdf.bounding_box([pdf.bounds.left, pdf.bounds.top], width: pdf.bounds.width) do
+          pdf.text('Vets.gov Submission', align: :right)
+        end
+      end
+    end
+
     def generate
       folder = 'tmp/pdfs'
       FileUtils.mkdir_p(folder)
       file_path = "#{folder}/extras_#{Time.zone.now}.pdf"
       generate_blocks = @generate_blocks
 
       Prawn::Document.generate(file_path) do |pdf|
-        generate_blocks.each do |block|
-          block.call(pdf)
+        generate_header(pdf)
```

Comment:
As @markolson mentioned, could this be part of the PDF stamping workflow task? Maybe update it to use this class if it has functionality you need?